### PR TITLE
Clean-up DPS service group enrollment sample and related documentation

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: azureiot.sln [hunk 1]
```diff
@@ -180,7 +180,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CleanupEnrollmentsSample",
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EnrollmentGroupSample", "provisioning\service\samples\getting started\EnrollmentGroupSample\EnrollmentGroupSample.csproj", "{EDE1FB8E-5BD4-49E7-9745-6599687D34CD}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "EnrollmentSample", "provisioning\service\samples\getting started\EnrollmentSample\EnrollmentSample.csproj", "{C8639C47-5772-4E58-A5A1-9D4134577437}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "IndividualEnrollmentSample", "provisioning\service\samples\getting started\EnrollmentSample\IndividualEnrollmentSample.csproj", "{C8639C47-5772-4E58-A5A1-9D4134577437}"
```
Comment (drwill-ms) [inline azureiot.sln:183]: Be more specific

File: provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs [hunk 2]
```diff
@@ -3,87 +3,88 @@
...
+        public EnrollmentGroupSample(ProvisioningServiceClient provisioningServiceClient)
         {
             _provisioningServiceClient = provisioningServiceClient;
-            _groupIssuerCertificate = groupIssuerCertificate;
         }
 
         public async Task RunSampleAsync()
         {
-            await QueryEnrollmentGroupAsync().ConfigureAwait(false);
+            await CreateEnrollmentGroupAsync();
```
Comment (drwill-ms) [inline provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs:23]: Create the enrollment before querying, to ensure there is at least one to query.

Don't use ConfigureAwait() in samples.

File: provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs [hunk 3]
```diff
@@ -3,87 +3,88 @@
...
-        public async Task GetEnrollmentGroupInfoAsync()
-        {
-            Console.WriteLine("\nGetting the enrollmentGroup information...");
-            EnrollmentGroup getResult =
-                await _provisioningServiceClient.GetEnrollmentGroupAsync(EnrollmentGroupId);
-            Console.WriteLine(getResult);
+            while (query.HasNext())
+            {
+                QueryResult queryResult = await query.NextAsync();
+                foreach (DeviceRegistrationState registration in queryResult.Items.Cast<DeviceRegistrationState>())
```
Comment (drwill-ms) [inline provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs:72]: Print more interesting information.

File: provisioning/service/samples/getting started/EnrollmentGroupSample/Program.cs [hunk 4]
```diff
@@ -36,14 +36,11 @@ public static int Main(string[] args)
                 Environment.Exit(1);
             }
 
-            var certificate = new X509Certificate2(parameters.CertificatePath);
-
             using var provisioningServiceClient = ProvisioningServiceClient.CreateFromConnectionString(parameters.ProvisioningConnectionString);
-            var sample = new EnrollmentGroupSample(provisioningServiceClient, certificate);
-            sample.RunSampleAsync().GetAwaiter().GetResult();
```
Comment (drwill-ms) [inline provisioning/service/samples/getting started/EnrollmentGroupSample/Program.cs:43]: Bad pattern.

File: provisioning/service/src/Config/QueryResult.cs [hunk 5]
```diff
@@ -77,6 +77,20 @@ public class QueryResult
         /// <summary>
         /// Getter for the list of query result Items.
         /// </summary>
+        /// <remarks>
+        /// Depending on the <see cref="Type"/>, these items can be cast to a corresponding type:
```
Comment (drwill-ms) [inline provisioning/service/src/Config/QueryResult.cs:81]: Document this on the item someone will be enumerating.

**Review by abhipsaMisra — APPROVED** [review]

**Review by schoims — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

**Pull request merged**